python/lsst/ip/diffim/zogy.py

```diff
@@ -898,3 +949,133 @@ class ZogyMapReduceConfig(ImageMapReduceConfig):
         doc='Zogy subtask to run on each sub-image',
         target=ZogyMapperSubtask
     )
+
+
+class ZogyImagePsfMatchConfig(ImagePsfMatchConfig):
+    zogyConfig = pexConfig.ConfigField(
+        dtype=ZogyConfig,
+        doc='ZogyTask config to use when running on complete exposure (non spatially-varying)',
+    )
+
+    zogyMapReduceConfig = pexConfig.ConfigField(
+        dtype=ZogyMapReduceConfig,
+        doc='ZogyMapReduce config to use when running Zogy on each sub-image (spatially-varying)',
+    )
+
+    def setDefaults(self):
+        self.zogyMapReduceConfig.gridStepX = self.zogyMapReduceConfig.gridStepY = 19
+        self.zogyMapReduceConfig.cellSizeX = self.zogyMapReduceConfig.cellSizeY = 20
+        self.zogyMapReduceConfig.borderSizeX = self.zogyMapReduceConfig.borderSizeY = 6
+        self.zogyMapReduceConfig.reducerSubtask.reduceOperation = 'average'
+
+
+class ZogyImagePsfMatchTask(ImagePsfMatchTask):
+    ConfigClass = ZogyImagePsfMatchConfig
+
+    def __init__(self, *args, **kwargs):
+        ImagePsfMatchTask.__init__(self, *args, **kwargs)
+
+    def setDefaults(self):
+        config = self.config.zogyMapReduceConfig
+        config.gridStepX = config.gridStepY = 19
+        config.cellSizeX = config.cellSizeY = 20
+        config.borderSizeX = config.borderSizeY = 6
+        config.reducerSubtask.reduceOperation = 'average'
+
+    def _computeImageMean(self, exposure):
+        """Compute the sigma-clipped mean of the pixels image of `exposure`.
+        """
+        statsControl = afwMath.StatisticsControl()
+        statsControl.setNumSigmaClip(3.)
+        statsControl.setNumIter(3)
+        ignoreMaskPlanes = ("INTRP", "EDGE", "DETECTED", "SAT", "CR", "BAD", "NO_DATA", "DETECTED_NEGATIVE")
+        statsControl.setAndMask(afwImage.MaskU.getPlaneBitMask(ignoreMaskPlanes))
+        statObj = afwMath.makeStatistics(exposure.getMaskedImage().getImage(),
+                                         exposure.getMaskedImage().getMask(),
+                                         afwMath.MEANCLIP|afwMath.MEDIAN, statsControl)
+        mn = statObj.getValue(afwMath.MEANCLIP)
+        med = statObj.getValue(afwMath.MEDIAN)
+        return mn, med
+
+    def subtractExposures(self, templateExposure, scienceExposure,
+                          doWarping=True, spatiallyVarying=True, inImageSpace=False,
+                          doPreConvolve=False):
+
```

Comment:
Unless I'm missing something that I did (wrong), I don't *think* it modifies the input exposures. I will surely add the docstring.